lib/NodeUtils.js

```diff
@@ -116,6 +123,65 @@ function attrname(a) {
   return a.name;
 }
 
+/**
+ * Escapes matching closing tag in a raw text.
+ *
+ * For example, given `<style>#text(</style><script></script>)</style>`,
+ * the parent tag would by "style" and the raw text is
+ * "</style><script></script>". If we come across a matching closing tag
+ * (in out case `</style>`) - replace `<` with `&lt;` to avoid unexpected
+ * and unsafe behavior after de-serialization.
+ */
+function escapeMatchingClosingTag(rawText, parentTag) {
+  const regexp = new RegExp('</' + parentTag, 'ig');
+
+  if (!regexp.test(rawText)) {
+    return rawText; // fast path
+  }
+
+  regexp.lastIndex = 0; // reset an index to start from the beginning of a string
+  const matches = rawText.matchAll(regexp);
+
+  let result = '';
+  let lastIndex = 0;
+  for (const match of matches) {
+    result += rawText.substring(lastIndex, match.index) + '&lt;';
+    // Skip over the `<` char, since we've replaced it above.
+    // Start the next iteration with a char following the `<` one.
+    lastIndex = match.index + 1;
+  }
+  // Add remaining contents of a string after the last `<` char.
+  if (lastIndex < rawText.length) {
+    result += rawText.substring(lastIndex);
+  }
+  return result;
+}
```

Comment:
On seconds thoughts this entire method can be simplified and written in a more performant way
```suggestion
function escapeMatchingClosingTag(rawText, parentTag) {
  if (!rawText.toLowerCase().includes('</'+ parentTag)) {
    return rawText; // fast path
  }

  const result = [...rawText];
  const matches = rawText.matchAll('</' + parentTag, 'ig');
  for (const match of matches) {
    result[match.index] = '&lt;';
  }

  return result.join('');
}

```